src/Form/SubscriptionForm.php

```diff
@@ -85,17 +85,20 @@ public function buildForm(array $form, FormStateInterface $form_state, stdClass
     // Build form according to received configuration:
     // Add contact fields.
     foreach ($profile->contact_fields as $contact_field_name => $contact_field) {
-      if ($contact_field['active']) {
-        $form[$contact_field_name] = array(
+      if (isset($contact_field['active']) && $contact_field['active']) {
+        $form['contact_fields'][$contact_field_name] = array(
           '#type' => Utils::getContactFieldType($contact_field),
           '#title' => $contact_field['label'],
           '#description' => $contact_field['description'],
           '#required' => !empty($contact_field['required']),
+          '#weight' => isset($contact_field['weight'])
+                    && filter_var($contact_field['weight'], FILTER_VALIDATE_INT) !== FALSE
+                    ? $contact_field['weight'] : 0,
```

Comment:
`filter_var()` already returns the filtered value, so this *could* be shortened to something like this:
```php
'#weight' => filter_var($contact_field['weight'] ?? NULL, FILTER_VALIDATE_INT) ?: 0,
```